tests/global_discovery_catalogue/test_gdc_functional.py

```diff
@@ -19,43 +19,178 @@
 
 TOPIC = 'cache/a/wis2/+/metadata/#'
 
 
+# fixtures
+
 @pytest.fixture
 def gb_client():
     options = {
-        'client_id': 'wis2-gdc-test-runner'
+        'client_id': 'wis2-gdc-test-runner',
+        'received_messages': []
     }
+
     return MQTTPubSubClient(os.environ['GB'], options)
 
 
+def subscribe_trigger_client():
+    options = {
+        'client_id': 'wis2-gdc-test-runner'
+    }
+
+    return MQTTPubSubClient(os.environ['TRIGGER_BROKER'], options)
+
+
+# paho-mqtt callbacks
+
 def _on_subscribe(client, userdata, mid, reason_codes, properties):
     client.subscribed_flag = True
 
 
+def _on_message(client, userdata, message):
+    client._userdata['received_messages'].append(message)
+
+
+def _disconnect_gb_client(gb_client_):
+    gb_client_.conn.loop_stop()
+    gb_client_.conn.disconnect()
+
+
+# helper functions
+
+def _subscribe_gb_client(gb_client_):
+    gb_client_.conn.subscribed_flag = False
+    gb_client_.conn.on_subscribe = _on_subscribe
+    gb_client_.conn.loop_start()
+
+    return gb_client_.conn.subscribe(TOPIC)
+
+
+def _publish_message(wcmp2_file, format_='trigger'):
+
+    base_url = 'https://raw.githubusercontent.com/wmo-im/wis2-global-services-testing/main/tests/global_discovery_catalogue'
+    md = None
+
+    with open(wcmp2_file) as fh:
+        md = fh.read()
+
+    if format_ == 'trigger':
+        message = {
+            'scenario': 'metadatatest',
+            'configuration': {
+                'setup': {
+                    'centreid': 11,
+                    'number': 1
+                },
+                'wnm': {
+                    'links': [{
+                        'href': f'{base_url}/{wcmp2_file}'
+                    }]
+                }
+            }
+        }
+
+    trigger_client = subscribe_trigger_client()
+    trigger_client.pub('config/a/wis2/metadata-pub', json.dumps(message))
+
+
 def test_global_broker_connection_and_subscription(gb_client):
+    print('Testing Global Broker connection and subscription')
+
     assert gb_client.conn.is_connected
-    gb_client.conn.subscribed_flag = False
-    gb_client.conn.on_subscribe = _on_subscribe
-    gb_client.conn.loop_start()
-    result, mid = gb_client.conn.subscribe(TOPIC)
+
+    result, _ = _subscribe_gb_client(gb_client)
+
     time.sleep(1)
+
     assert result is mqtt.MQTT_ERR_SUCCESS
     assert gb_client.conn.subscribed_flag
-    gb_client.conn.loop_stop()
-    gb_client.conn.disconnect()
+
+    _disconnect_gb_client(gb_client)
 
 
 def test_notification_and_metadata_processing_success(gb_client):
-    pass
+    print('Testing Notification and metadata processing (success)')
+
+    assert gb_client.conn.is_connected
+
+    result, _ = _subscribe_gb_client(gb_client)
+
+    time.sleep(1)
+
+    assert result is mqtt.MQTT_ERR_SUCCESS
+    assert gb_client.conn.subscribed_flag
+
+    _publish_message('global_discovery_catalogue/valid/urnwmomdca-eccc-mscweather.observations.swob-realtime.json')
```

Comment:
Based on other changes, shouldn't it be `io-wis2dev-11-test` and not `ca-eccc-msc` ?

Reply:
Fixed in subsequent commit.